wallet/wallet.h

```diff
@@ -215,6 +215,25 @@ static inline enum wallet_payment_status wallet_payment_status_in_db(enum wallet
 	fatal("%s: %u is invalid", __func__, w);
 }
 
+/* Types of transactions we store in the `transactions` table. Mainly used for
+ * display purposes later. */
+enum wallet_tx_type {
+       TX_UNKNOWN = 0,
+       TX_THEIRS = 1,  /* This only affects their funds in the channel */
+       TX_WALLET_DEPOSIT = 2,
+       TX_WALLET_WITHDRAWAL = 4,
+       TX_CHANNEL_FUNDING = 8,
+       TX_CHANNEL_CLOSE = 16,
+       TX_CHANNEL_UNILATERAL = 32,
```

Comment:
We don't distinguish their unilateral from ours?

Reply:
Ah, I see, THEIRS is a bit, the rest are distinct.  Maybe we should just make those three cases distinct?

TX_CHANNEL_OUR_UNILATERAL / TX_CHANNEL_THEIR_UNILATERAL, TX_CHANNEL_OUR_HTLC_SUCCESS ... TX_CHANNEL_OUR_HTLC_TIMEOUT...?

That gets us closer to the onchain types, too.

Reply:
Not really, a close transaction is also a deposit to us if we had funds in them, but annotate at different places, creating all possible combinations is likely not what we want.

But I agree that `TX_THEIRS` is a bit strange since it indicates both who originated the transaction as well as filtering out transactions we don't have a stake in (e.g., their sweeps).